tools/azure-sdk-tools/ci_tools/conda/CondaConfiguration.py

```diff
@@ -0,0 +1,196 @@
+
+http = urllib3.PoolManager()
+# arguments: |
+#       -c "${{ replace(convertToJson(parameters.CondaArtifacts), '"', '\"') }}"
+#       -w "$(Build.SourcesDirectory)/conda/conda-recipes"
+#
+# # Sample configuration yaml. This is converted to json before being passed a blob to the invoking script. (see directly above for conversion)
+#     - name: uamqp
+#         common_root: uamqp
+#         in_batch: true
+#         conda_py_versions:
+#         - "38"
+#         - "39"
+#         - "310"
+#         - "311"
+#         checkout:
+#         - package: uamqp
+#         download_uri: https://files.pythonhosted.org/packages/0b/d8/fc24d95e6f6c80851ae6738c78da081cd535c924b02c5a4928b108b9ed42/uamqp-1.6.5.tar.gz
+#   - name: azure-core
+#     common_root: azure
+#     in_batch: ${{ parameters.release_azure_core }}
+#     channels: "conda-forge"
+#     checkout:
+#     - package: azure-core
+#       version: 1.24.0
+#     - package: azure-core
+#       version: 1.24.0
+#   - name: azure-storage
+#     common_root: azure
+#     in_batch: ${{ parameters.release_azure_storage }}
+#     checkout:
+#     - package: azure-storage-blob
+#       checkout_path: sdk/storage
+#       version: 12.12.0
+#     - package: azure-storage-queue
+#       checkout_path: sdk/storage
+#       version: 12.3.0
+#     - package: azure-storage-file-share
+#       checkout_path: sdk/storage
+#       version: 12.8.0
+#     - package: azure-storage-file-datalake
+#       checkout_path: sdk/storage
+#       version: 12.7.0
+
+def get_pypi_page(package: str, version: str) -> bs4.BeautifulSoup:
+    url = f"https://pypi.org/project/{package}/{version}/"
+
+    try:
+        r = http.request("GET", url)
+    except Exception as e:
+        raise RuntimeError(f"We must get this the data {url} to continue proper assembly.")
+
+    return bs4.BeautifulSoup(r.data.decode("utf-8"), "html.parser")
+
+
+def get_package_sdist_url(package: str, version: str) -> str:
+    soup = get_pypi_page(package, version)
+
+    # source distribution always first on the page
+    try:
+        target_zip = soup.select("div.file__card a")[0]["href"]
+    except Exception as e:
+        print(f"Cant get data from {package} and {version}")
+        raise e
+    filename = os.path.basename(target_zip)
+
+    if filename.endswith("tar.gz") or filename.endswith(".zip"):
+        return target_zip
+    else:
+        raise ValueError(f"Unable to find a source distribution for {package}@{version}.")
+
+
+class CheckoutConfiguration:
+    def __init__(self, raw_json: dict):
+        # we should always have a package name
+
+        if "package" in raw_json:
+            self.package = raw_json["package"]
+        else:
+            raise ValueError("A checkout configuration MUST have a package name defined in key 'package'.")
+
+        if "checkout_path" in raw_json:
+            self.checkout_path = raw_json["checkout_path"]
+        else:
+            self.checkout_path = None
```

Comment:
These assignments look like they could be simplified with `.get()`:
```suggestion
        self.checkout_path = raw_json.get("checkout_path")
```

Reply:
I left my brain in a jar back in Duvall 👍